packages/mco/components/mco-dashboard/data-policy/parsers/subscription-parser.tsx

```diff
@@ -0,0 +1,187 @@
+import * as React from 'react';
+import { APPLICATION_TYPE, DRPC_STATUS } from '@odf/mco/constants';
+import {
+  DisasterRecoveryResourceKind,
+  SubscriptionGroupType,
+  useSubscriptionResourceWatch,
+} from '@odf/mco/hooks';
+import { ACMPlacementModel } from '@odf/mco/models';
+import {
+  ACMManagedClusterKind,
+  ACMPlacementRuleKind,
+  DRClusterAppsMap,
+  DRClusterKind,
+  ProtectedAppsMap,
+} from '@odf/mco/types';
+import {
+  findDRType,
+  getClustersFromDecisions,
+  getProtectedPVCsFromDRPC,
+} from '@odf/mco/utils';
+import { ApplicationKind } from '@odf/shared';
+import { getName, getNamespace } from '@odf/shared/selectors';
+import * as _ from 'lodash-es';
+
+const createProtectedApp = (
+  application: ApplicationKind,
+  applicationNameSpace: string,
+  clusterName: string,
+  subscriptionGroupsList: SubscriptionGroupType[]
+): ProtectedAppsMap => {
+  const protectedApp: ProtectedAppsMap = {
+    appName: getName(application),
+    appNamespace: applicationNameSpace,
+    appKind: application?.kind,
+    appAPIVersion: application?.apiVersion,
+    appType: APPLICATION_TYPE.SUBSCRIPTION,
+    placementInfo: [],
+  };
+
+  subscriptionGroupsList.forEach((subscriptionGroup) => {
+    const { drInfo } = subscriptionGroup;
+    if (!_.isEmpty(drInfo)) {
+      const {
+        drPlacementControl,
+        drPolicy,
+        drClusters: currentDRClusters,
+      } = drInfo;
+
+      const placementInfo = {
+        deploymentClusterName: clusterName,
+        drpcName: getName(drPlacementControl),
+        drpcNamespace: getNamespace(drPlacementControl),
+        protectedPVCs: getProtectedPVCsFromDRPC(drPlacementControl),
+        replicationType: findDRType(currentDRClusters),
+        syncInterval: drPolicy?.spec?.schedulingInterval,
+        workloadNamespace: applicationNameSpace,
+        failoverCluster: drPlacementControl?.spec?.failoverCluster,
+        preferredCluster: drPlacementControl?.spec?.preferredCluster,
+        lastGroupSyncTime: drPlacementControl?.status?.lastGroupSyncTime,
+        status: drPlacementControl?.status?.phase as DRPC_STATUS,
+      };
+
+      protectedApp.placementInfo.push(placementInfo);
+    }
+  });
```

Comment:
```suggestion
  subscriptionGroupsList.forEach((subscriptionGroup) => {
    const { drInfo } = subscriptionGroup;
    if (!_.isEmpty(drInfo)) {
      const {
        drPlacementControl,
        drPolicy,
        drClusters: currentDRClusters,
      } = drInfo;

      const placementInfo = {
        deploymentClusterName: clusterName,
        drpcName: getName(drPlacementControl),
        drpcNamespace: getNamespace(drPlacementControl),
        protectedPVCs: getProtectedPVCsFromDRPC(drPlacementControl),
        replicationType: findDRType(currentDRClusters),
        syncInterval: drPolicy?.spec?.schedulingInterval,
        workloadNamespace: applicationNameSpace,
        failoverCluster: drPlacementControl?.spec?.failoverCluster,
        preferredCluster: drPlacementControl?.spec?.preferredCluster,
        lastGroupSyncTime: drPlacementControl?.status?.lastGroupSyncTime,
        status: drPlacementControl?.status?.phase as DRPC_STATUS,
      };

      protectedApp.placementInfo.push(placementInfo);
    }
  });
  
```

move this part as a separate function called createPlacementInfoList